Extension to solidity framework

rin-st opened the pull request:
- rename extensions to solidityFramework (option, folder, vars, types, docs etc). `solidityFramework` option became a single value instead of `extensions` array
- removed `config.json` logic and readme since as I understand it's not used at all

Fixes #67

technophile-04 commented:
Hey Rinat could just fix the merge conflicts I think they were caused due to #55

rin-st commented:
Thanks! Fixed

technophile-04 reviewed (commented):
Thanks @rin-st !! This is really nice!! 

I was also since we are now clear we are no more having any extensions in repo, maybe remove : 

https://github.com/scaffold-eth/create-eth/blob/4db51acd26de1a1af888feaff6bcc599948cd791/src/config.ts#L10


I think we can also remove this logic : 

https://github.com/scaffold-eth/create-eth/blob/170bc6d94c51e291cdfc35d5476e248c4e4339e9/src/utils/prompt-for-missing-options.ts#L33-L52

and simplify this logic too, since things are now deterministic : 

https://github.com/scaffold-eth/create-eth/blob/170bc6d94c51e291cdfc35d5476e248c4e4339e9/src/types.ts#L42-L62

rin-st commented on src/tasks/copy-template-files.ts:
```diff
@@ -111,31 +117,31 @@ const copyExtensionsFiles = async ({ dev: isDev }: Options, extensionPath: strin
-        extensionTemplatePath => ({
+  const solidityFrameworkTemplatedFileDescriptors: TemplateDescriptor[] = solidityFrameworkPath
+    ? findFilesRecursiveSync(solidityFrameworkPath, filePath => isTemplateRegex.test(filePath))
+        .map(extensionTemplatePath => ({
```
updated, thanks! [cfd9e58](https://github.com/scaffold-eth/create-eth/pull/68/commits/cfd9e58152a052827e77d3355bc195f88f1b16d0)

rin-st commented on src/tasks/copy-template-files.ts:
```diff
@@ -111,31 +117,31 @@ const copyExtensionsFiles = async ({ dev: isDev }: Options, extensionPath: strin
+  templateDir: string,
   targetDir: string,
 ) => {
+  const solidityFrameworkPath = getSolidityFrameworkPath(solidityFramework, templateDir);
```
thanks! Updated [5543001](https://github.com/scaffold-eth/create-eth/pull/68/commits/5543001509c032f89628dd63bafbf3441e9973c9)

rin-st commented on src/tasks/copy-template-files.ts:
```diff
@@ -276,19 +281,19 @@ const setUpExternalExtensionFiles = async (options: Options, tmpDir: string) =>
-      await copyExtensionsFiles(options, extensionPath, targetDir);
-    }),
-  );
+  let solidityFrameworkPath;
```
[5543001](https://github.com/scaffold-eth/create-eth/pull/68/commits/5543001509c032f89628dd63bafbf3441e9973c9) now it's `const`, but still out of `if` since it's used later

rin-st commented on src/utils/consts.ts:
```diff
@@ -1,4 +1,4 @@
-export const baseDir = "base";
+export const BASE_DIR = "base";
```
Yes! Changed it, also 
- moved `BASE_DIR` to `copy-template-files` since it used only here.
- renamed `EXTERNAL_EXTENSION_TMP_FOLDER` to `EXTERNAL_EXTENSION_TMP_DIR` similar to other consts

rin-st commented on src/tasks/copy-template-files.ts:
```diff
@@ -308,7 +313,7 @@ export async function copyTemplateFiles(options: Options, templateDir: string, t
 
   // 4. Process templated files and generate output
-  await processTemplatedFiles(options, basePath, targetDir);
+  await processTemplatedFiles(options, basePath, templateDir, targetDir);
```
Good point! [cd2e630](https://github.com/scaffold-eth/create-eth/pull/68/commits/cd2e630fb9785d6f2f8bdda590b3619097b94351)

rin-st commented:
Thank you for thorough review Shiv!

Regarding your last message

> I was also since we are now clear we are no more having any extensions in repo, maybe remove :

> I think we can also remove this logic :

> and simplify this logic too, since things are now deterministic :

I removed some types, simplified prompt logic and changed `config.ts` to `curated-extensions.ts` since there's nothing more in that file except curated extensions [4fd327b](https://github.com/scaffold-eth/create-eth/pull/68/commits/4fd327b4fbc1aadea8237837633072133358a174)

technophile-04 reviewed (approved):
Really awesome work Rinat!! Tested everything throughly out (solidity frameworks, external extension, curated extension, + dev mode) and everything works nicely!! 

Just last small thing could update this too? : 

https://github.com/scaffold-eth/create-eth/blob/d476ed9945b7b866ee51f147aa9f0f8f91412af2/contributors/core-extensions.md#L3-L21

rin-st commented:
> Just last small thing could update this too? :
> 
> https://github.com/scaffold-eth/create-eth/blob/d476ed9945b7b866ee51f147aa9f0f8f91412af2/contributors/core-extensions.md#L3-L21

Good eye 👍! 

[2d9c67e](https://github.com/scaffold-eth/create-eth/pull/68/commits/2d9c67e7c173a1f222c9d8b71dbd58347106f8bd) Removed the whole text since we don't have extensions other than solidity framework and config is also removed.

> Just a quick note to mention that adding `null` as an item in the `extensions` property will show the "None" option to the user. That option doesn't add any extension for the given question. 

Thought about this part but since we have it only in one place (not even exact same way) https://github.com/scaffold-eth/create-eth/blob/e6034a42d7989bd5588958f99396273282652e77/src/utils/prompt-for-missing-options.ts#L34 and it's clear how it works so I think we don't need this text too

---

Renamed `core-extensions.md` to `TEMPLATING.md` since file became mostly about templating

technophile-04 commented:
> and it's clear how it works so I think we don't need this text too
> Renamed core-extensions.md to TEMPLATING.md since file became mostly about templating

Yup yup 100% agree! 

Thanks @rin-st!!! Merging this!
